modules/reactor/include/meshgenerators/PolygonMeshGeneratorBase.h

```diff
@@ -65,18 +65,49 @@ class PolygonMeshGeneratorBase : public MeshGenerator
     HEXAGON_NUM_SIDES = 6
   };
 
+  /// A struct type used to contain multiple blocks's boundary layer related parameters
+  struct multiBdryLayerParams
+  {
+    std::vector<Real> widths;
+    std::vector<Real> fractions;
+    std::vector<unsigned int> intervals;
+    std::vector<Real> biases;
+  };
+  /// A struct type used to contain a single block's boundary layer related parameters
+  struct singleBdryLayerParams
+  {
+    Real width;
+    unsigned int intervals;
+    Real bias;
+  };
+
 protected:
   /**
    * Creates a mesh of a slice that corresponds to a single side of the polygon to be generated.
    * @param mesh input mesh to build the slice mesh onto
    * @param ring_radii radii of the ring regions
-   * @param rings numbers of radial intervals of the ring regions
+   * @param ring_layers numbers of radial intervals of the ring regions
+   * @param ring_radial_biases values used for radial meshing biasing in ring regions
```

Comment:
does the BdryLayerParam not work for regular ring regions?

Reply:
regular regions does not need width or fraction attribute.